sway/desktop/xdg_shell_v6.c

```diff
@@ -210,6 +224,18 @@ static void handle_destroy(struct wl_listener *listener, void *data) {
 	view_destroy(&xdg_shell_v6_view->view);
 }
 
+static void handle_request_fullscreen(struct wl_listener *listener, void *data) {
+	struct sway_xdg_shell_v6_view *xdg_shell_v6_view =
+		wl_container_of(listener, xdg_shell_v6_view, request_fullscreen);
+	struct wlr_xdg_toplevel_v6_set_fullscreen_event *e = data;
+
+	if (xdg_shell_v6_view->view.wlr_xdg_surface_v6->role != WLR_XDG_SURFACE_V6_ROLE_TOPLEVEL) {
```

Comment:
This should probably be a sway_assert

Reply:
Unless the protocol allows you to set popups as fullscreen surfaces... I don't think it does